etcdserver/etcdhttp/client_auth_test.go

```diff
@@ -238,6 +238,50 @@ func TestAuthFlow(t *testing.T) {
 			wcode: http.StatusOK,
 			wbody: ``,
 		},
+		{
+			req: (func() *http.Request {
+				req := mustJSONRequest(t, "DELETE", "enable", "")
+				req.SetBasicAuth("root", "bad")
+				return req
+			})(),
+			store: mockAuthStore{
+				enabled: true,
+				user: &auth.User{
+					User:     "root",
+					Password: goodPassword,
+					Roles:    []string{"root"},
+				},
+				roles: map[string]*auth.Role{
+					"root": {
+						Role: "root",
+					},
+				},
+			},
+			wcode: http.StatusUnauthorized,
+			wbody: `{"message":"Insufficient credentials"}`,
+		},
+		{
+			req: (func() *http.Request {
+				req := mustJSONRequest(t, "DELETE", "enable", "")
+				req.Header.Set("Authorization", "malformedencoding")
+				return req
+			})(),
+			store: mockAuthStore{
+				enabled: true,
+				user: &auth.User{
+					User:     "root",
+					Password: goodPassword,
+					Roles:    []string{"root"},
+				},
+				roles: map[string]*auth.Role{
+					"root": {
+						Role: "root",
+					},
```

Comment:
not sure I understand. Use guest role for the bad password test case or use guest role for malformed basic auth encoding test case?